--- a/chan_dongle.c
+++ b/chan_dongle.c
@@ -116,7 +116,7 @@
 		basename = devname;
 
 	/* NOTE: use system system wide lock directory */
-	return snprintf(buf, length, "/var/lock/LCK..%s", basename);
+	return snprintf(buf, length, "/opt/var/lock/LCK..%s", basename);
 }
 
 #/* return 0 on error */
--- a/tools/tty.c
+++ b/tools/tty.c
@@ -44,7 +44,7 @@
 		basename = devname;
 
 	/* TODO: use asterisk build settings for /var/lock */
-	return snprintf(buf, length, "/var/lock/LCK..%s", basename);
+	return snprintf(buf, length, "/opt/var/lock/LCK..%s", basename);
 }
 
 #/* return 0 on error */
